js/apps/admin-ui/src/clients/authorization/ResourcesPolicySelect.tsx

```diff
@@ -108,57 +128,141 @@ export const ResourcesPolicySelect = ({
     [search],
   );
 
+  const [toggleUnsavedChangesDialog, UnsavedChangesConfirm] = useConfirmDialog({
+    titleKey: t("unsavedChangesTitle"),
+    messageKey: t("unsavedChangesConfirm"),
+    continueButtonLabel: t("common:continue"),
+    continueButtonVariant: ButtonVariant.danger,
+    onConfirm: () => {
+      navigate(
+        toPolicyDetails({
+          realm: realm,
+          id: clientId,
+          policyId: clickedPolicy?.id!,
+          policyType: clickedPolicy?.type!,
+        })
+      );
+    },
+  });
+
   const toSelectOptions = () =>
     items.map((p) => (
       <SelectOption key={p.id} value={p.id}>
         {p.name}
       </SelectOption>
     ));
 
+  const toChipGroupItems = (
+    field: ControllerRenderProps<PolicyRepresentation, Type>
+  ) => {
+    return (
+      <ChipGroup>
+        {field.value?.map((permissionId) => {
+          const policy = items.find(
+            (permission) => permission.id === permissionId
+          );
+
+          if (!policy) {
+            return null;
+          }
+
+          const to = toPolicyDetails({
+            realm: realm,
+            id: clientId,
+            policyId: policy.id!,
+            policyType: policy.type!,
+          });
+
+          return (
+            <Chip
+              key={policy.id}
+              onClick={(event) => {
+                event.stopPropagation();
+                field.onChange(field.value?.filter((id) => id !== policy.id));
+              }}
+            >
+              {policy.type ? (
+                <Link
+                  to={to}
+                  onClick={(event) => {
+                    if (isDirty) {
+                      event.preventDefault();
+                      event.stopPropagation();
```

Comment:
Do we have to stop propagation? I believe preventing the default behaviour here should be enough, but I can be wrong.

Reply:
IMO its needed for following case:

![grafik](https://github.com/keycloak/keycloak/assets/69153350/08148b32-8d97-4e32-b017-677bb24647c7)

When you click on a Chip-Link while the SelectOptions are shown, your first click would just hide the SelectOptions (even when you click on the Dialogs Continue/Cancel buttons). That's why I added a `stopPropagation()`. Preventing the default behavior just assures that the `onclick` does its thing and hinders the `<Link to={to}>` of immediately navigating somewhere to.

Reply:
Ahhh, yes this makes perfect sense. Consider it resolved!